Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -857,14 +857,7 @@ public class GraphTraverser: GraphTraversing {
                     return false
                 }
             },
-            skip: { dependency in
-                switch dependency {
-                case .xcframework:
-                    return false
-                case .framework, .library, .bundle, .packageProduct, .target, .sdk:
-                    return true
-                }
-            }
+            skip: { $0.isDynamicPrecompiled }
```

Comment:
In this method we're trying to collect all xcframeworks transitively 

XCFrameworkA -> XCFrameworkB -> ...

The skip clause determines when to terminate the traversal of a specific graph path, an xcframework can be dynamic and is precompiled as such in some cases the traversal here will terminate earlier than we want it to (e.g. if XCFrameworkB is dynamic) 

In this method the goal is to skip non xcframeworks so it was correct before. 

A change may be needed if the following can ever be possible (somehow via Tuist cache logic, but it's not possible via public API). 

XCFrameworkA -> dynamicLibA.dylib -> XCFrameworkB 

XCFrameworkA -> PrecompiledFrameworkA.framework -> XCFrameworkB 

In which case the skip clause would be revised to not skip precompiled libraries and frameworks - but this need a bit more testing to verify.